[Hsien Kit] iP

KHsienKit opened the pull request.

KHsienKit committed:
Update gitignore to include class files and modified Chatbot name to V

KHsienKit committed:
Fix bug to include .class files in gitignore

KHsienKit committed:
Make V echo commands

KHsienKit committed:
Add list functionality to V

KHsienKit committed:
Add mark as done function to V

Created a new class called Task to hold the description and whether the task is done or not
V can now parse commands and check the first word to see what kind of command to execute

KHsienKit committed:
Clean up code to fit coding standards

KHsienKit committed:
Change if else statements to switch case

KHsienKit committed:
Add .vscode folder to gitignore

KHsienKit committed:
Add tracking ToDos, Events and Deadlines

ZiliaAJY commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,19 @@
+public class Deadline extends Task {
+    private final String TYPE = "D";
+    private String by;
```
Consider declaring attributes as protected rather than private as it would be easier to facilitate extension of subclasses as the project progress.

ZiliaAJY commented on src/main/java/V.java:
```diff
@@ -0,0 +1,96 @@
+    }
+
+    public static void greet() {
+        String logo = " _       _ \n"
+                + "\\ \\     / / \n"
+                + " \\ \\   / / \n"
+                + "  \\ \\_/ / \n"
+                + "   \\___/ \n";
```
Consider to declare as public static final

ZiliaAJY commented on src/main/java/V.java:
```diff
@@ -0,0 +1,96 @@
+        boolean isOnline = true;
+        Task[] listOfTasks = new Task[100];
+        int count = 0;
+        String temp;
```
variable name "temp" can be more descriptive.

ZiliaAJY commented on src/main/java/V.java:
```diff
@@ -0,0 +1,96 @@
+            line = input.nextLine();
+            lineArr = line.trim().split(" ");
+            switch (lineArr[0]) {
+            case "bye":
+                input.close();
+                isOnline = false;
+                break;
```
Good use of boolean flag isOnline as it allows the program to have a clear flow and makes the code easy to understand.

ZiliaAJY reviewed (commented):
Good job!

KHsienKit committed:
Change starting programme from V.java to Main.java

shaunngoh commented on src/main/java/V.java:
```diff
@@ -0,0 +1,96 @@
+            }
+        }
+        printBlock("See Ya");
+    }
```
You could consider shortening your main method by breaking down the respective tasks into smaller methods

shaunngoh commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,25 @@
+public class Event extends Task {
+    private final String TYPE = "E";
```
Good use of declaring a constant to determine the type of task when returning from the getType method!

KHsienKit committed:
Shift deadline case statement above event

shaunngoh commented on src/main/java/V.java:
```diff
@@ -0,0 +1,96 @@
+        String to;
+        String line;
+        String[] lineArr;
+        Scanner input = new Scanner(System.in);
```
The respective variables here are hard to follow, maybe you could provide a comment for each of the variable's use case?

shaunngoh commented on src/main/java/V.java:
```diff
@@ -0,0 +1,96 @@
+        String[] lineArr;
+        Scanner input = new Scanner(System.in);
+        
+        while (isOnline) {
```
Good naming convention for your boolean variable!

shaunngoh reviewed (commented):
Overall, I found that your code was easy to read, though maintainability might be slightly harder due to the complex nature of your main function. Goodjob!

KHsienKit committed:
Clean up code to meet quality standards

KHsienKit committed:
Add exception to mark and deadline function

A NumberFormatException is thrown when the user does not input a valid integer when marking a task
A InvalidDeadlineException is thrown when the user does not add a "/by" or a deadline after the "/by"

KHsienKit committed:
Merge branch 'branch-Level-5'

KHsienKit committed:
Add comment line to trigger new commit for new tag

KHsienKit committed:
Use ArrayList<Task> to keep track list of tasks

KHsienKit committed:
Automate text ui testing

KHsienKit committed:
Clean up unwanted comments

KHsienKit committed:
Add support for deleting tasks from the list

KHsienKit committed:
Remove comment line from previous commit

That comment line was meant for master branch, not branch-Level-7

KHsienKit committed:
Remove comment line to add A-Exceptions tag

KHsienKit committed:
Add getType() method for all child class of Task

KHsienKit committed:
Add V.txt to .gitignore

KHsienKit committed:
Add getter functions for Task and its subclasses

KHsienKit committed:
Change file run and compilation name to Main

Initially the file name was Duke and the programme could not be compiled

KHsienKit committed:
Add save functionality

KHsienKit committed:
Merge branch 'branch-Level-6'

KHsienKit committed:
Merge branch 'branch-Level-7'

KHsienKit committed:
Fix bugs from merge with branch-Level-7

KHsienKit committed:
Add META-INF folder and .jar file to gitignore

KHsienKit committed:
Fix loadSaveFile bug

There was an error loading an event into the list. This bug was caused by an improper save of events where the "/by" parameter was used instead of "/to" in the V.txt
This caused an exception when V tried to load the save file

KHsienKit committed:
Fix delete task bug

Delete task had a bug where number of task left after delete was +1 the actual number of task

KHsienKit committed:
Add compile script

KHsienKit committed:
Create folder to hold compile script and files

Folder "local-compile" was created to store all class files and jar file from compiling V and also to store the script file to compile V

KHsienKit committed:
Move compile.bat file out of folder

File was moved out of the folder and folder was deleted as keeping a singular file in a folder was redundant

KHsienKit committed:
Create txt file to hold path of all .java files

KHsienKit committed:
Add META-INF folder to class-files folder

KHsienKit committed:
Remove sources.txt file

KHsienKit committed:
Implement .bat file to compile and make jar file

KHsienKit committed:
Fix bug where class files were added to a folder

Originally, class files were added to a folder called "class-files" but that caused an error where the jar file could not run
class-files folder was removed and compile.bat now saves the class files in the same directory

KHsienKit committed:
Initialise Storage Class

KHsienKit committed:
Add saveFilePath parameter to Storage class method

KHsienKit committed:
Initiliase Parser class

KHsienKit committed:
Initiliase TaskList class

KHsienKit committed:
Initiliase Ui class

KHsienKit committed:
Add command to delete save file of previous run

KHsienKit committed:
Extract out TaskList class

Created a new class called TaskList to handle list functions

KHsienKit committed:
Fix empty list display bug

V will no longer display an empty list if the save file was empty

KHsienKit committed:
Extract storage class

KHsienKit committed:
Fix bug where V print to terminal during load save

V was printing to terminal when loading the save file. This bug was fixed by adding a boolean isFromSaveFile parameter to the add tasks function where if the task was added from the save file, the function will not print the output of the addition of the task

KHsienKit committed:
Extract clearSaveFile method

KHsienKit committed:
Reorganise code in Storage.java for clearer read

KHsienKit committed:
Add constructor for Storage with filepath

KHsienKit committed:
Add Parser class

KHsienKit committed:
Add Ui class

KHsienKit committed:
Add parser to TaskList when loading save file

KHsienKit committed:
Add find command for parser

KHsienKit committed:
Add print line seperator function to Ui class

KHsienKit committed:
Add find function to TaskList

KHsienKit committed:
Add JavaDoc to all classes and functions

KHsienKit committed:
Merge pull request #1 from KHsienKit/branch-Level-9

Branch level 9

KHsienKit committed:
Merge branch 'master' into branch-A-JavaDoc

KHsienKit committed:
Merge pull request #2 from KHsienKit/branch-A-JavaDoc

Merge branch-A-JavaDoc to master

KHsienKit committed:
Add case insensitive search for find function

KHsienKit committed:
Add user guide

KHsienKit committed:
Add approriate line breaks to README

KHsienKit committed:
Add whitespace for intro

KHsienKit committed:
Fix typo in README

KHsienKit committed:
Add cutoff line at the bottom of README

KHsienKit committed:
Remove "##" from bottom of README

KHsienKit committed:
Add error handling for add event, mark and delete

KHsienKit committed:
Extract printing of error message to Ui class

KHsienKit committed:
Add link to final release to README

KHsienKit committed:
Fix README error on save file location

okkhoy commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,84 @@
+     */
+    public boolean parse(String command, boolean isFromSaveFile) {
+        try {
+            wordArray = command.trim().split(" ");
+            switch (wordArray[0].toLowerCase()) {
```
Can SLAP this out.

okkhoy commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,84 @@
+                taskList.deleteTask(position);
+                return true;
+            case "todo":
+                description = String.join(" ", Arrays.copyOfRange(wordArray, 1, wordArray.length));
```
can SLAP this out; see the example from previous lecture.

okkhoy commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,105 @@
+     * Loads save file using file path saved in <code>saveFilePath</code> attribute.
+     * Stores all the commands as a String in an array.
+     */
+    public void loadSave() {
```
the name can be better. reason: does `loadSave` mean load (read from file) or save (write to file)?

okkhoy commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,105 @@
+     * A new file is created if it a save file does not exist.
+     * @param taskList list of tasks
+     */
+    public void createSave(TaskList taskList) {
```
same comment as above

okkhoy commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,105 @@
+            for (Task task: list) {
+                switch(task.getType()) {
+                case "T":
+                    saveFileWriter.write("todo " + task.getDescription() + System.lineSeparator());
```
you do have a toString() method, right? can try to us that here.

okkhoy commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    /**
+     * @return ArrayList of String containing all the commands from save file
```
better: Returns the ArrayList of string ...

okkhoy reviewed (commented):
In general, code is well written. Try to organize the code as per A-Packages for better code management. Pay attention to (some of the) method names.
